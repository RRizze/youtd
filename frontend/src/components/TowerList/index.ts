export { default } from "./tower-list";
