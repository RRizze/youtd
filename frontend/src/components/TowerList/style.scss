.towers-list {
  padding-top: 20px;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
}

.towers-list__item {
  padding: 35px 1% 0px;
  max-width: 880px;
  width: 50%;
  display: block;
}