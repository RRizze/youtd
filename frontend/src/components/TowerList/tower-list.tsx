import React from "react";
import Tower from "../Tower";
import "./style.scss";
import { ITowerDTO } from "../../models/tower";

export interface Props { towers: ITowerDTO[]; }

const TowerList = ({ towers }: Props) => (
  <ul className="towers-list">
    {towers.map((tower) => (
      <li key={tower.id} className="towers-list__item">
        <Tower {...tower} />
      </li>
    ))}
  </ul>
);

export default TowerList;
