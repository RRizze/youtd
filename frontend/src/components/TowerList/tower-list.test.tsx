import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import TowerList from "./tower-list";

describe("TowerList tests", () => {
  it("render correctly", () => {
    const props = {
      towers: [{
      id: 1,
      tower_name: "shrub",
      goldcost: 30,
      icon: "../images/shrub.png",
      wave_lvl_up: 0,
      element_lvl_up: 0,
      range_tower: 800,
      min_damage: 30,
      max_damage: 30,
      attack_cd: 0.7,
      attack_type_name: "",
      element_name: "",
      rarity_name: "",
      tower_description: "",
      skills: "",
    }]};
    const tree = renderer
      .create(<TowerList {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("render with props", () => {
    const props = {
      towers: [{
      id: 1,
      tower_name: "shrub",
      goldcost: 30,
      icon: "../images/shrub.png",
      wave_lvl_up: 0,
      element_lvl_up: 0,
      range_tower: 800,
      min_damage: 30,
      max_damage: 30,
      attack_cd: 0.7,
      attack_type_name: "",
      element_name: "",
      rarity_name: "",
      tower_description: "",
      skills: "",
    }]},
      wrapper = mount(<TowerList {...props} />);

    expect(wrapper.props()).toEqual(props);
  });
});
