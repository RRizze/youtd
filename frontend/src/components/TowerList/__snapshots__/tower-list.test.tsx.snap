// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TowerList tests render correctly 1`] = `
<ul
  className="towers-list"
>
  <li
    className="towers-list__item"
  >
    <div
      className="tower"
      data-id={1}
    >
      <div
        className="tower__general"
      >
        <div
          className="tower__header"
        >
          <figure
            className="tower__img"
          >
            <img
              alt="tower icon"
              className="tower__icon"
              src="../images/shrub.png"
            />
            <figcaption>
              <h2
                className="tower__name"
              >
                shrub
              </h2>
            </figcaption>
          </figure>
        </div>
        <div
          className="tower__features"
        >
          <table
            className="tower__table"
          >
            <thead>
              <tr>
                <th
                  className="tower__table-header"
                  colSpan={2}
                >
                  General Info
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <th>
                  Element
                </th>
                <td
                  style={
                    Object {
                      "color": "undefined",
                    }
                  }
                >
                  
                </td>
              </tr>
              <tr>
                <th>
                  Attack Type
                </th>
                <td>
                  
                </td>
              </tr>
              <tr>
                <th>
                  Rarity
                </th>
                <td
                  style={
                    Object {
                      "color": "undefined",
                    }
                  }
                >
                  
                </td>
              </tr>
              <tr>
                <th>
                  Goldcost
                </th>
                <td>
                  30
                </td>
              </tr>
              <tr>
                <th>
                  Wave Lvl
                </th>
                <td>
                  0
                </td>
              </tr>
              <tr>
                <th>
                  Element Lvl
                </th>
                <td>
                  0
                </td>
              </tr>
            </tbody>
          </table>
          <table
            className="tower__table"
          >
            <thead>
              <tr>
                <th
                  className="tower__table-header"
                  colSpan={2}
                >
                  Combat stats
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <th>
                  DPS
                </th>
                <td>
                  21
                </td>
              </tr>
              <tr>
                <th>
                  Range
                </th>
                <td>
                  800
                </td>
              </tr>
              <tr>
                <th>
                  Cd
                </th>
                <td>
                  0.7
                </td>
              </tr>
              <tr>
                <th>
                  Damage
                </th>
                <td>
                  30-30
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div
        className="tower__main"
      >
        <div
          className="tower__description"
        >
          <h2
            className="tower__description-title"
          >
            Tower Description:
          </h2>
          <p>
            
          </p>
        </div>
        <div
          className="tower__abilities"
        >
          <h2
            className="tower__abilities-title"
          >
            Abilities:
          </h2>
        </div>
      </div>
    </div>
  </li>
</ul>
`;
